### DISCUSSION TITLE
How to make an selectable buttons in telegram

### DISCUSSION THREAD

**Discussion by Darvin02** [botman/botman#1373] [Q&A]
I was trying to make clickable buttons that, when clicked, would not open a new menu or keyboard, but that the button would become "clicked" on an already open keyboard. Can anyone tell me how to implement this?
<img width="389" alt="image" src="https://github.com/botman/botman/assets/88323253/4b4630db-a8ca-445b-8d36-7dc06fc238f5">
<img width="462" alt="image" src="https://github.com/botman/botman/assets/88323253/1be54df9-0ce2-499b-ada9-55fa8ee2d83c">

**Comment by filippotoso-goodmood** [comment]


**Reply by Darvin02** [reply]
Telegram driver

**Reply by filippotoso** [reply]
Is it something possible in the Telegram Bot API?
Can you provide a link to the official documentation about it?

**Reply by Darvin02** [reply]
In the attached screenshots you can see how it looks like.
Here it is, as i think, documentation about it:
https://core.telegram.org/bots/api#inlinekeyboardmarkup

**Reply by filippotoso** [reply]
Try this: https://gist.github.com/yarkm13/4d9359f2c96149a012f563ea41849c02

**Reply by Darvin02** [reply]
Unfortunately this is not the case, the keyboard must be Inline format, with several buttons and when you click on the button, this button is highlighted as in the attached screenshots

**Reply by filippotoso** [reply]
If you are not able to adapt that code to fits your need, you can always build your own payload and use the additionalParameters field

**Reply by Darvin02** [reply]
Unfortunately, I can not figure out how to do it in general and would be very grateful if someone could help at least with an example of how to do it.
